log-collector/src/main/scala/io/phdata/pulse/logcollector/Util.scala

```diff
@@ -0,0 +1,34 @@
+package io.phdata.pulse.logcollector
+
+import io.phdata.pulse.common.domain.LogEvent
+
+object Util {
+
+  /**
+   * Convert a [[LogEvent]] to a flattened [[Map[String,String]]
+   * @param logEvent LogEvent to convert
+   * @return The event, with properties flattened, as a [[Map[String,String]]
+   */
+  def logEventToFlattenedMap(logEvent: LogEvent): Map[String, String] = {
+    val values = logEvent.productIterator
+    val logEventmap = logEvent.getClass.getDeclaredFields.map {
+      _.getName -> (values.next() match {
+        case x => x
+      })
+    }.toMap
+    logEventmap
+      .flatten {
+        case ((key, map: Option[Map[String, String]]))
+            if map.getOrElse(None).isInstanceOf[Map[String, String]] =>
+          map.get
+        case ((key, value: Option[String])) if value.getOrElse(None).isInstanceOf[String] =>
+          Map(key -> value.getOrElse(None))
+        case ((key, value)) => Map(key -> value)
+      }
+      .toMap
+      .filter(kv => !kv._2.equals(None))
```

Comment:
Is it possible to not add these in the first place?

Reply:
Going to rewrite per you other comment about LogEvent's fields always being known.

Reply:
New "hardcoded" function still does this.  Lets discuss further once I push the changes and you get to take a look.

Reply:
Can the above expressions be `getOrElse("")` instead of `getOrElse(None)`? Then it's an empty string and no check for `None` is needed